### PR TITLE
GTC-2081: Fix log message for >26 input bands; cache input band enumeration

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Specifying >26 input bands ALMOST works... but doesn't because of a rogue log statement.

Issue Number: GTC-2081

## What is the new behavior?

-Specifying >26 input bands ACTUALLY works

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by manukala6 — APPROVED** [review]

File: gfw_pixetl/layers.py [hunk 1]
```diff
@@ -219,8 +218,9 @@ def _input_bands(self) -> List[List[InputBandElement]]:
                     )
 
                 for i in range(file_band_count):
+                    band_name: str = enumerate_bands(i + 1)[-1]
                     LOGGER.info(
-                        f"Adding {file_uri} (band {i+1}) to input band {string.ascii_uppercase[len(input_bands) + i]}"
+                        f"Adding {file_uri} (band {i+1}) as input band {band_name}"
```
Comment (gtempus) [inline gfw_pixetl/layers.py:223]: That reads lots better! 👓

File: gfw_pixetl/utils/utils.py [hunk 2]
```diff
@@ -224,6 +225,18 @@ def union(
...
+    corresponding to numbers in base 26.
+
+    Taken from https://stackoverflow.com/a/29351603.
+    """
+    for size in itertools.count(1):
+        for letters in itertools.product(string.ascii_uppercase, repeat=size):
+            yield "".join(letters)
+
+
+@lru_cache(typed=False)
```
Comment (gtempus) [inline gfw_pixetl/utils/utils.py:239]: This is cool! I didn't know about this decorator. What caused you to add it, @dmannarino? Did you notice something in testing, or did it get flagged while doing some profiling? Just curious. 🧠

**Review by gtempus — APPROVED** [review]
Thanks, @dmannarino!

File: gfw_pixetl/utils/utils.py (hunk 2 above)
Comment (dmannarino) [reply gfw_pixetl/utils/utils.py:239]: Nope, I realized that it was going to get called over and over again with the same value, and it's computationally expensive (well, A LITTLE). Possibly erring on the side of premature optimization, but innocuous enough.

**Pull request merged**